lib/gcloud/pubsub.rb

```diff
@@ -43,14 +56,168 @@ def self.pubsub project = nil, keyfile = nil
   end
 
   ##
-  # Google Cloud Pubsub
+  # = Google Cloud Pubsub
+  #
+  # Google Cloud Pub/Sub is designed to provide reliable, many-to-many,
+  # asynchronous messaging between applications. Publisher applications can
+  # send messages to a "topic" and other applications can subscribe to that
+  # topic to receive the messages. By decoupling senders and receivers, Google
+  # Cloud Pub/Sub allows developers to communicate between independently written
+  # applications.
+  #
+  # Gcloud's goal is to provide a API that is familiar and comfortable to
+  # Rubyists. Authentication is handled by Gcloud.pubsub. You can provide the
+  # project and credential information to connect to the Pubsub service, or if
+  # you are running on Google Compute Engine this configuration is taken care
+  # of for you.
+  #
+  #   require "glcoud/pubsub"
+  #
+  #   pubsub = Gcloud.pubsub
+  #
+  #   topic = pubsub.topic "my-topic"
+  #   topic.publish "task completed"
+  #
+  # To learn more about Datastore, read the
+  # {Google Cloud Pubsub Overview
+  # }[https://cloud.google.com/pubsub/overview].
+  #
+  # == Retrieving Topics
+  #
+  # A Topic is a named resource to which messages are sent by publishers.
+  # A Topic is found by its name. (See Project#topic)
+  #
+  #   require "glcoud/pubsub"
+  #
+  #   pubsub = Gcloud.pubsub
+  #
+  #   topic = pubsub.topic "my-topic"
+  #
+  # == Creating a Topics
+  #
+  # A Topic is created from a Project. (See Project#create_topic)
+  #
+  #   require "gcloud/pubsub"
+  #
+  #   pubsub = Gcloud.pubsub
+  #
+  #   topic = pubsub.create_topic "my-topic"
+  #
+  # == Publishing Messages
+  #
+  # Messages are published to a topic. (See Topic#publish)
+  #
+  #   require "glcoud/pubsub"
+  #
+  #   pubsub = Gcloud.pubsub
+  #
+  #   topic = pubsub.topic "my-topic"
+  #   msg = topic.publish "new-message"
+  #
+  # Messages can also be published with attributes:
+  #
+  #   require "glcoud/pubsub"
+  #
+  #   pubsub = Gcloud.pubsub
```

Comment:
Can we condense these a bit? Feels like they're extra spaced apart -- almost needlessly so.